src/Npgsql/TypeHandlers/NumericHandlers/NumericHandler.cs

```diff
@@ -72,7 +130,7 @@ public override decimal Read(NpgsqlReadBuffer buf, int len, FieldDescription fie
 
             var overflow = false;
 
-            var result = 0M;
+            var result = new DecimalRaw() { Value = 0M };
```

Comment:
You can remove `{ Value = 0M }` because it's already initialized to 0. Use `new DecimalRaw()` or `default(DecimalRaw)`.